content/docs/architecture/resources/virtualmachinetemplate.md

```diff
@@ -1,49 +1,91 @@
-    config_map: {"key": "value", "another_key": ... [] ...}
-``` 
-
-
-A VM template is merely a representation of a possible VM implementation by a provider. For example, in AWS this template may correspond to the creation of a `t3.medium` whereas in Azure it may correspond to `Standard_D2_v2`. It is up to each `Environment` to determine how to implement a `VirtualMachineTemplate`.
+  config_map:
+    cloud-config: ""
+    image: ubuntu-20-04-x64
+    size: s-1vcpu-512mb-10gb
+    ssh_username: root
+    webinterfaces: '[]'
+  image: ubuntu-20-04-x64
+  name: do-ubuntu-20-04-x64
+```
 
 ## Configuration
 ### `name`
-
 Display name for the VM template.
 
 ### `image`
+Default value for the image to use when provisioning this VM.
 
-Default value for the image to use when provisioning this VM. Acceptable values will depend upon the provisioner in use. Useful either as a default value for a default provisioner, or when multiple environments use the same provisioner. Can be overridden via configuration in the `Environment` resource. 
+Acceptable values will depend upon the provisioner in use. Useful either as a default value for a default provisioner, or when multiple environments use the same provisioner. Can be overridden via configuration in the `Environment` resource.
 
 ### `config_map`
-The ConfigMap provides the ability to customize more values given to the provisioner. Useful if the provisioner can distinguish between Regions or different sizing options. 
+The ConfigMap provides the ability to customize more values given to the provisioner. Useful if the provisioner can distinguish between regions or different sizing options.
 
-An example for a configmap vor VMs on Hetzner could look the following 
+An example for a `config_map` for virtual machines on [Hetzner](https://www.hetzner.com) could look the following 
 
 ```yaml
-server_type:   cx21 # Hetzner has available sizings cx11, cx21, cx31 ...
-ssh_username:  root # Default SSH Username is root
-location:      nbg1 # Region
+## Example Hetzner config_map
+spec:
+  config_map:
+    server_type: cx21   ## Hetzner has available sizings cx11, cx21, cx31 ...
+    ssh_username: root  ## Default SSH Username is root
+    location: nbg1      ## Region
 ```
 
 ## Cloud-Init configuration
-`VirtualMachineTemplates` may offer Cloud-Init configuration. Currently this is saved under the `cloud-config` key inside the `config_map`.
-This is supposed to be changed to be saved in it's own field inside the VMT. 
+`VirtualMachineTemplates` may offer [Cloud-Init](https://cloudinit.readthedocs.io/en/latest/) configuration. Currently this is saved under the `cloud-config` key inside the `config_map`.
+
+```yaml
+## Example Cloud-Init configuration
+spec:
+  config_map:
+    cloud_config: |
+      runcmd:
+        - echo "Hello World"
+```
+
+> **NOTE:** Future releases of HobbyFarm will allow the Cloud-Init configuration to be saved in it's own field inside the VirtualMachineTemplate.
```

Comment:
It seems more prudent to me to save this in a configmap than in the CRD itself. 

Just noting that here for posterity? I guess.

Reply:
@ebauman I'll update to include that it's fluid and cloud be either/or.